add a way to disable PTY allocation

TomasTomecek opened the pull request:
Fixes #179

This patch resolves the original issue: https://github.com/ansible/ansible/pull/26170#issuecomment-312591504

Signed-off-by: Tomas Tomecek <ttomecek@redhat.com>

rhatdan commented on docs/buildah-run.md:
```diff
@@ -14,6 +14,12 @@ the *buildah config* command.
 
 ## OPTIONS
 
+**--tty**
```
How about 
```
**--tty**

By default a pseudo-TTY is allocated only when buildah's standard output is
attached to a pseudo-TTY.  Setting the `--tty` option to `true` will cause a
pseudo-TTY to be allocated inside the container. Setting the `--tty` option to `false` will prevent the pseudo-TTY from being allocated.
```

nalind replied:
... standard output is attached to a terminal.  Though we might want to change it to point to check standard input instead.

TomasTomecek replied:
I like your suggestions.

rhatdan commented:
Do we need the same feature for buildah bud?


We also need bash completion.

TomasTomecek commented:
> Do we need the same feature for buildah bud?

`docker build` doesn't have such option. I suspect it may not be needed. The only problem may be if some tools try to be interactive (that's what I run into with `run`). Not sure if PTY is allocated by default.

I think I addressed all the comments.

rhatdan commented:
LGTM

rhatdan commented:
@nalind Do you want to merge?

TomSweeneyRedHat commented:
LGTM

nalind commented:
@rhatdan Yup, looks good to me.  Merging.

nalind commented:
@rh-atomic-bot r+ 62b79edc26c95ac85dfe898af91ad8145ed90377
